src/main/scala/org/broadinstitute/dsde/workbench/sam/service/UserService.scala

```diff
@@ -233,13 +233,14 @@ class UserService(val directoryDAO: DirectoryDAO, val cloudExtensions: CloudExte
   }
 
   private def enableIdentityIfTosAccepted(user: WorkbenchUser, samRequestContext: SamRequestContext): IO[Unit] = {
+    val gracePeriodEnabled = tosService.tosConfig.enableGracePeriod
     tosService.getTosStatus(user.id)
       .flatMap {
-        // If the user has accepted TOS or TOS is disabled, then enable the user in LDAP
-        // If the user is an SA, it's also acceptable to enable them in LDAP
+        // If the user has accepted TOS, the grace period is enabled, or TOS is disabled, then enable the user in LDAP
+        // Additionally, if the user is an SA, it's also acceptable to enable them in LDAP
         case Some(true) | None => registrationDAO.enableIdentity(user.id, samRequestContext)
         case _ =>
-          if(isServiceAccount(user.email.value)) registrationDAO.enableIdentity(user.id, samRequestContext)
+          if(isServiceAccount(user.email.value) || gracePeriodEnabled) registrationDAO.enableIdentity(user.id, samRequestContext)
```

Comment:
How long is the grace period? Where is the code that makes sure that the user must accept the ToS once the grace period has expired for them?

Reply:
It's an undefined amount of time. We're really only implementing this grace period to support a smoother rollout for AoU. They indicated that for this initial rollout of ToS, they need the APIs to be enabled in production but for it to not cause any breakage for users, so this is our solution to that. Once AoU gives us the go-ahead, we'll flip the boolean in sam.conf which will end the grace period. Later on, this grace period code can be removed.

Reply:
Oh, I see. Is there a ticket to remove the grace period code when it is no longer needed?

Reply:
Not yet! I will make one